src/app/modules/querybuilder/components/querybuilder-editor/edit/edit-criterion/edit-criterion.component.ts

```diff
@@ -62,13 +72,72 @@ export class EditCriterionComponent implements OnInit, AfterViewChecked {
     }
 
     this.showGroups = this.query.groups.length > 1
+
+    if (!this.featureService.mockLoadnSave()) {
+      this.loadUIProfile()
+    }
+  }
+
+  ngOnDestroy(): void {
+    this.subscriptionCritProfile?.unsubscribe()
   }
 
   ngAfterViewChecked(): void {
     this.actionDisabled = this.isActionDisabled()
     this.changeDetector.detectChanges()
   }
 
+  loadUIProfile(): void {
+    this.subscriptionCritProfile?.unsubscribe()
+    const version = this.criterion.termCodes[0].version
+      ? '&version=' + this.criterion.termCodes[0].version
+      : ''
+    const param =
+      'code=' +
+      this.criterion.termCodes[0].code +
+      '&system=' +
+      this.criterion.termCodes[0].system +
+      version
+    this.subscriptionCritProfile = this.backend
+      .getTerminologyProfile(param)
+      .subscribe((profile) => {
```

Comment:
@thkoehler11  - missing error handling if profile cannot be found.
Backend throws a 500 internal server error if a profile cannot be found - it should be caught here and handled accordingly.

To reproduce:
delete a profile from the database table `public.ui_profile_table` and select it in the UI to add to a query.
Should throw the following error in the console:
<img width="1507" alt="image" src="https://user-images.githubusercontent.com/18329952/182130530-5ca53750-b66e-44a2-827b-61e48b7a7658.png">